spec/beaker/dsl/roles_spec.rb

```diff
@@ -163,6 +163,14 @@ class ClassMixedWithDSLRoles
       agent1[:type] = 'aio-foss'
       expect( subject.aio_version?(agent1) ).to be === true
     end
+    it 'can take an empty string for pe_ver' do
+      agent1[:pe_ver] = ''
+      expect{ subject.aio_version?(agent1) }.not_to raise_error
+    end
+    it 'can take an empty string for FOSS version' do
+      agent1[:version] = ''
+      expect{ subject.aio_version?(agent1) }.not_to raise_error
+    end
```

Comment:
In looking deeper I see *some* tests for some of the cases in the prior tests. I think this could be made more rigorous -- i.e., step through all the possibilities, in truth-table fashion:

| pe_ver | version | aio_version?|
|-----------|----------|------------------|
| nil | nil | false |
| '' | nil | false |
| nil | '' | false |
| '' | '' | false |
| 3.8 | nil | false |
| 3.8 | '' | false | 
| etc. | etc. | etc. |
|-----|----|----|

There's probably a way to built a simple list of lists and just run through the assertions that way.  It would provide a lot better confidence in the behavior, and I wouldn't be surprised if it didn't turn up a couple of bugs.  Adding in the role and type strings to the matrix would probably be useful.

If this seems like drudgery, I can pair with you on it. :grinning: